Source/Generic/ECTestComparisons.m

```diff
@@ -53,6 +53,32 @@ - (NSString*)nameForMatching
 	return @"number";
 }
 
+// Returns YES if both numbers are doubles which only differ in the last couple of digits.
+// This could happen if we convert a double to a string and then re-interpret it.
+- (BOOL)matchesAsNearAsDamnIt:(NSNumber*)other {
+	if ( strcmp(self.objCType,"d") == 0 && strcmp(other.objCType,"d") == 0) {
+		double a = self.doubleValue;
+		double b = other.doubleValue;
+		return (fabs(a - b) <= ((fabs(a) > fabs(b)) ? fabs(b) : fabs(a)) * DBL_EPSILON * 4.0);
+	} else {
+		return NO;
+	}
+}
+
+- (BOOL)matches:(id)item2 context:(NSString*)context level:(NSUInteger)level block:(ECTestComparisonBlock)block {
```

Comment:
As mentioned elsewhere, I don't like the fact that this has made the float comparisons less accurate for all tests, and not just ones where it is specifically needed.

I would prefer to add an `options` parameter which can enable/disable the fuzzy testing.

As luck would have it, I think that @SheffieldKevin has implemented just such a thing in another issue...